src/BookingPriceEvaluator.php

```diff
@@ -0,0 +1,159 @@
+use Dhii\I18n\StringTranslatingTrait;
+use Dhii\Iterator\CountIterableCapableTrait;
+use Dhii\Iterator\ResolveIteratorCapableTrait;
+use Dhii\Storage\Resource\SelectCapableInterface;
+use Dhii\Util\Normalization\NormalizeIntCapableTrait;
+use Dhii\Util\Normalization\NormalizeStringCapableTrait;
+use InvalidArgumentException;
+use Psr\Container\ContainerInterface;
+use RebelCode\Bookings\BookingInterface;
+
+/**
+ * Evaluates booking prices.
+ *
+ * @since [*next-version*]
+ */
+class BookingPriceEvaluator implements EvaluableInterface
+{
+    /* @since [*next-version*] */
+    use CountIterableCapableTrait;
+
+    /* @since [*next-version*] */
+    use ResolveIteratorCapableTrait;
+
+    /* @since [*next-version*] */
+    use ContainerGetCapableTrait;
+
+    /* @since [*next-version*] */
+    use NormalizeKeyCapableTrait;
+
+    /* @since [*next-version*] */
+    use NormalizeIntCapableTrait;
+
+    /* @since [*next-version*] */
+    use NormalizeStringCapableTrait;
+
+    /* @since [*next-version*] */
+    use NormalizeContainerCapableTrait;
+
+    /* @since [*next-version*] */
+    use CreateContainerExceptionCapableTrait;
+
+    /* @since [*next-version*] */
+    use CreateNotFoundExceptionCapableTrait;
+
+    /* @since [*next-version*] */
+    use CreateInvalidArgumentExceptionCapableTrait;
+
+    /* @since [*next-version*] */
+    use CreateOutOfRangeExceptionCapableTrait;
+
+    /* @since [*next-version*] */
+    use CreateRuntimeExceptionCapableTrait;
+
+    /* @since [*next-version*] */
+    use StringTranslatingTrait;
+
+    /**
+     * The services SELECT resource model.
+     *
+     * @since [*next-version*]
+     *
+     * @var SelectCapableInterface
+     */
+    protected $servicesSelectRm;
+
+    /**
+     * The expression builder.
+     *
+     * @since [*next-version*]
+     *
+     * @var object
+     */
+    protected $exprBuilder;
+
+    /**
+     * Constructor.
+     *
+     * @since [*next-version*]
+     *
+     * @param SelectCapableInterface $servicesSelectRm The services SELECT resource model.
+     * @param object                 $exprBuilder      The expression builder.
+     */
+    public function __construct(SelectCapableInterface $servicesSelectRm, $exprBuilder)
+    {
+        $this->servicesSelectRm = $servicesSelectRm;
+        $this->exprBuilder      = $exprBuilder;
+    }
+
+    /**
+     * {@inheritdoc}
+     *
+     * @since [*next-version*]
+     */
+    public function evaluate(ValueAwareInterface $ctx = null)
+    {
+        $booking = $ctx->getValue();
+
+        if (!($booking instanceof BookingInterface)) {
+            throw $this->_createRuntimeException(
+                $this->__('Cannot determine price - argument is not a booking instance'), null, null
+            );
+        }
+
+        try {
+            $booking = $this->_normalizeContainer($booking);
+        } catch (InvalidArgumentException $exception) {
+            throw $this->_createRuntimeException(
```

Comment:
Not allowed here. Did we agree that for now `RuntimeException` will be thrown due to absence of base exception implementations?

Reply:
Yes